docs/plugin/interfaces/guidelines.md

```diff
@@ -0,0 +1,383 @@
+|0xA0000000|0xFFFFFFFF|QA Interfaces|```ThunderInterfaces/interfaces/ids.h```|
+
+Tags/annotations available for optimizing the generated code {#TagsAnnotationsAvailableForOptimizingTheGeneratedCode}
+============================================================
+ 
+Tags/annotations that can be used to optimize the generated code, influence the generated JSON-RPC interface from the interface (if desired at all), or document the interface can be found here:
+ 
+See: [https://rdkcentral.github.io/Thunder/plugin/interfaces/tags](https://rdkcentral.github.io/Thunder/plugin/interfaces/tags/)
+
+Practical Considerations {#PracticalConsiderations}
+========================
+
+Less is more {#LessIsMore}
+------------
+
+When designing an interface spend some time thinking on how the interface can be as easy to use, understand and can be consistent without the need of adding a lot of documentation and the need of explaining all the cornercases that can happen when using the interface in a certain way.
+
+As an interface is in principle immutable it is better to spend more time on this before releasing the interface than later on trying to improve the interface or writing a lot of documentation explaining on how to use it in certain situations. 
+
+When designing the interface it is good to use the "less it more" principle when adding methods to it to make sure it can do what should be possible with it. 
+Less methods will make it easier to understand the interface, less documentation will be needed, it is easier to keep it consistent and there is less need to think or document what will happen if you use the methods in an unexpected order. 
+
+Let's design an Player Interface as an example. At first one would expect all the following should be possible for a player:
+
+* Start playback
+* Stop Playback
+* Pause Playback
+* Rewind (with a certain speed perhaps)
+* Forward (with a certain speed perhaps)
+
+So a naive interface could look like this:
+
+``` c++
+struct EXTERNAL IPlayer : public virtual Core::IUnknown {
+    enum { ID = ID_PLAYER };
+
+    core::hresult Play() const = 0;
+    core::hresult Stop() = 0;
+    core::hresult Pause() const = 0;
+    core::hresult Rewind(const uint32_t speed) = 0;
+    core::hresult Forward(const uint32_t speed) = 0;
+};
+```
+At first this might seem logical but how would you expect the implementation to respond, if you would do Pause(), before Play()? Or Play() after Rewind() without stopping first? You would need to think about all the different options and specify how the interface should respond in that situation. 
+But if you think of it, what all the different methods do is specify a certain playback speed. So if you have one method with which you could set the player speed (and allow a negative number) it could do all the above while at the same time be consistent, easy to understand and no need to explain what will happen in certain situation as just the last set speed is the one that is current.
```

Comment:
While I fully agree and endorse the "less is more" principle, often it is contextual on "how much is less". Clarity of the interface gets reduced after certain threshold on "less". In my opinion, we might want to make this sounds like a "follow the balanced approach".